app/controlplane/pkg/biz/workflowrun.go

```diff
@@ -388,6 +395,50 @@ func (uc *WorkflowRunUseCase) GetByIDInOrg(ctx context.Context, orgID, runID str
 	return wfRun, nil
 }
 
+type VerificationResult struct {
+	Result        bool
+	FailureReason string
+}
+
+func (uc *WorkflowRunUseCase) Verify(ctx context.Context, run *WorkflowRun) (*VerificationResult, error) {
```

Comment:
what would happen if there is no bundle, i.e an old attestation?

Reply:
It would just skip the verification, but I've added an additional check to have full control.